merlin/dag/graph.py

```diff
@@ -133,16 +133,22 @@ def remove_inputs(self, to_remove):
 
         while nodes_to_process:
             node, columns_to_remove = nodes_to_process.popleft()
-
             if node.input_schema and len(node.input_schema):
                 output_columns_to_remove = node.remove_inputs(columns_to_remove)
 
                 for child in node.children:
-                    nodes_to_process.append((child, to_remove + output_columns_to_remove))
+                    nodes_to_process.append(
+                        (child, list(set(to_remove + output_columns_to_remove)))
```

Comment:
This line's change is just to de-duplicate the list of output columns. without it, we would end up with something like `["target", "target"]` that would still technically work but made less sense when debugging this.